x-pack/plugins/actions/server/actions_client.ts

```diff
@@ -92,13 +97,17 @@ export class ActionsClient {
     this.actionExecutor = actionExecutor;
     this.executionEnqueuer = executionEnqueuer;
     this.request = request;
+    this.authorization = authorization;
   }
 
   /**
    * Create an action
    */
-  public async create({ action }: CreateOptions): Promise<ActionResult> {
-    const { actionTypeId, name, config, secrets } = action;
+  public async create({
+    action: { actionTypeId, name, config, secrets },
+  }: CreateOptions): Promise<ActionResult> {
+    await this.authorization.ensureAuthorized('create', actionTypeId);
```

Comment:
This is true, but only as a side effect, and I'd rather be explicit about using the actions_authorization here for a couple of reasons:

1. The next stage in Actions RBAC is to add `producer` level privileges, like Alerts, and at that point we'll need this anyway. This will likely be in 7.10 as well or 7.11, so not long.
2. Using the same audit logger means the messages will be phrased the same, whether they're an unauthorized request to do `create` (which would use the SO one) or `execute` (which would use the actions one). The fact that under the hood they're different, shouldn't affect that.
3. From a maintainability perspective - keeping a uniform shape to all the different operations, so that they all approach this in the same way seems valuable and makes it easier to maintain the code with multiple developers coming in and out of it.

Anyway, that's my reasoning but if you have an objection, I'm all ears :)

Reply:
Thanks for the explanation. My primary concerns for the current approach are:
1) Duplicate authorization checks - in reality, I wouldn't expect this to have a perceivable performance impact, but it's hard to say for sure.
2) Duplicate audit records - both the `actions_client` and the Secure Saved Objects Client will end up recording these access attempts, which both adds noise to the log, and makes it harder to reason about what actually happened.

Is there a reason to use to the secured saved objects client here, as opposed to the unsecure version? If you're gating all actions behind your own authorization checks, _and_ you're doing your own audit logging, then I don't see a real benefit to using the secured instance. Moving to the unsecure instance would alleviate both of my concerns here, as we'd be removing both types of duplication.

^ I will warn you that I am pre-coffee, so there could very likely be a legitimate reason for keeping the secure version around that I'm overlooking